assets/src/exit-modal-survey/index.js

```diff
@@ -10,78 +9,76 @@ import './style.css';
 import { data } from '@eventespresso/eejs';
 import { stringify } from 'querystringify';
 
-const { exitModali18n: i18n = {}, exitModalInfo: info = {}} = data;
+const { exitModali18n: i18n = {}, exitModalInfo: info = {} } = data;
 
-const queryString = stringify({
-    firstname: info.firstname,
-    emailaddress: info.emailaddress,
-    website: info.website
-});
+const queryString = stringify( {
+	firstname: info.firstname,
+	emailaddress: info.emailaddress,
+	website: info.website,
+} );
 
 const modalProps = {
-    styles: {
-        content: {
-            display: 'flex',
-            top: '80px',
-            left: '50%',
-            bottom: 'auto',
-            right: 'auto',
-            WebkitTransform: 'translateX(-50%)',
-            transform: 'translateX(-50%)'
-        },
-        typeFormStyle: {
-            width: '600px',
-            height: '400px'
-        }
-    },
-    typeFormUrl: 'https://eventespresso.typeform.com/to/O1DDym?' + queryString,
-    introText: i18n.introText,
-    doSurveyButtonText: i18n.doSurveyButtonText,
-    skipButtonText: i18n.skipButtonText,
-    buttonClass: {
-        doSurvey: 'button button-primary',
-        closeModal: 'button button-secondary'
-    }
+	styles: {
+		content: {
+			display: 'flex',
+			top: '80px',
+			left: '50%',
+			bottom: 'auto',
+			right: 'auto',
+			WebkitTransform: 'translateX(-50%)',
+			transform: 'translateX(-50%)',
+		},
+		typeFormStyle: {
+			width: '600px',
+			height: '400px',
+		},
+	},
+	typeFormUrl: 'https://eventespresso.typeform.com/to/O1DDym?' + queryString,
+	introText: i18n.introText,
+	doSurveyButtonText: i18n.doSurveyButtonText,
+	skipButtonText: i18n.skipButtonText,
+	buttonClass: {
+		doSurvey: 'button button-primary',
+		closeModal: 'button button-secondary',
+	},
 };
 
 /**
  * Handler for deactivation trigger
  * @param {Event} e
  * @param {element} el
  */
-const handleDeactivationClick = (e, el) => {
-    e.preventDefault();
-    const link = $(el).attr('href');
-    const appContainer = document.getElementById('ee-exit-survey-modal');
-    let modalContainer;
-    ReactModal.setAppElement(appContainer);
-    modalContainer = ReactDOM.render(<ExitModal {...modalProps} />, appContainer);
-    modalContainer.el.addEventListener(
-        CLOSE_MODAL_EVENT,
-        function(e){
-            window.location.href = link;
-        },
-        {once: true}
-    );
+const handleDeactivationClick = ( e, el ) => {
+	e.preventDefault();
+	const link = $( el ).attr( 'href' );
+	const appContainer = document.getElementById( 'ee-exit-survey-modal' );
+	ReactModal.setAppElement( appContainer );
+	const modalContainer = ReactDOM.render( <ExitModal { ...modalProps } />, //eslint-disable-line react/no-render-return-value
```

Comment:
take note of the way you can disable linting for a specific line.  In this case, I disabled the linting for this line because although it's not recommended to use the return value of `ReactDOM.render`, I'm using it in a way that is supposed to be safe (not the way which gave rise to the warning).  So the linting doesn't apply in this case.